content/en/blog/_posts/2022-11-28-kubelet-credential-providers/index.md

```diff
@@ -0,0 +1,69 @@
+---
+layout: blog
+title: 'Kubernetes v1.26: GA Support for Kubelet Credential Providers'
+date: 2022-11-28
+slug: kubelet-credential-providers
+---
+
+**Authors:** Andrew Sy Kim (Google), Dixita Narang (Google)
+
+Kubernetes v1.26 introduces GA support for Kubelet Credential Providers,
+offering an extensible plugin framework to dynamically fetch credentials for any container registry service.
+
+## Background
+
+Kubernetes supports the ability to dynamically fetch credentials for a container registry service.
+However, prior to Kubernetes v1.20, this capability was compiled into the Kubelet and only available for
+Amazon Elastic Container Registry, Azure Container Registry, and Google Cloud Container Registry.
+
+{{< figure src="kubelet-credential-providers-in-tree.png" alt="" >}}
+
+Figure 1: Kubelet Built-in Credential Provider Support for Amazon Elastic Container Registry, Azure Container Registry, and Google Cloud Container Registry.
+
+Kubernetes v1.20 introduced alpha support for Kubelet Credential Providers, providing a mechanism for
+the Kubelet to dynamically authenticate and pull images for arbitrary container registry services.
+In Kubernetes v1.26, this feature is now GA.
+
+{{< figure src="kubelet-credential-providers-plugin.png" alt="" >}}
+
+Figure 2: Kubelet Credential Provider Overview
+
+## Why is it Important?
+
+Prior to Kubernetes v1.20, it was required to modify Kubelet to add support to dynamically fetch credentials
+for image registries other than ACR (Azure Container Registry), ECR (Elastic Container Registry), and GCR (Google Container Registry).
+Kubelet Credential Provider provides an extensible plugin mechanism that can be used by any cloud provider
+without adding built-in logic into the kubelet.
+
+## How it Works
+
+The Kubelet and plugin communicate through stdio (stdin, stdout, and stderr) by sending and receiving json-serialized api-versioned types.
+If the exec plugin is enabled and the Kubelet requires authentication information for an image that matches against a plugin,
+the Kubelet will exec the plugin, passing CredentialProviderRequest API via stdin. Then the exec plugin communicates with the
+image registry to dynamically fetch the credentials and returns the credentials in an encoded response of the CredentialProviderResponse
+API to the Kubelet via stdout.
```

Comment:
(nits)
“kubelet” (all lower case)

`CredentialProviderRequest` and `CredentialProviderResponse` (monospace)